src/Symfony/Component/Security/Http/Authentication/AuthenticatorManager.php

```diff
@@ -60,6 +60,12 @@ class AuthenticatorManager implements AuthenticatorManagerInterface, UserAuthent
      */
     public function __construct(iterable $authenticators, TokenStorageInterface $tokenStorage, EventDispatcherInterface $eventDispatcher, string $firewallName, LoggerInterface $logger = null, bool $eraseCredentials = true, bool $hideUserNotFoundExceptions = true, array $requiredBadges = [])
     {
+        foreach ($authenticators as $authenticator) {
```

Comment:
Traversing `$authenticators` breaks laziness. The check should be done as late as possible, just before calling `supports()` on the authenticator.

Reply:
Thanks, updated